src/battle_main.c

```diff
@@ -3744,6 +3744,9 @@ static void DoBattleIntro(void)
             gBattleStruct->overworldWeatherDone = FALSE;
             SetAiLogicDataForTurn(AI_DATA); // get assumed abilities, hold effects, etc of all battlers
             Ai_InitPartyStruct(); // Save mons party counts, and first 2/4 mons on the battlefield.
+            #if B_DYNAMIC_MOVE_DESCRIPTIONS == TRUE
+                SetMoveUIDataForTurn(MOVE_UI_DATA);
+            #endif
```

Comment:
what I would do is move all those call to `Ai_InitPartyStruct` since this one will be always code it saves some lines